### ISSUE TITLE
Component: Tree

### ISSUE THREAD

**Issue by mlmoravek** [oruga-ui/oruga#1500]
### Description

Since the current OMenu component functions more like a tree component, the OMenu component should be converted into an OTree component which properly implement the W3C accessibility [Tree View Pattern](https://www.w3.org/WAI/ARIA/apg/patterns/treeview/).

Then, the component Menu can be reimplemented to reflect a more specific menu and menu bar component.

### Why Oruga need this feature

Use a component name that better reflects the component's functionality.
